webrender/src/prim_store.rs

```diff
@@ -314,20 +287,13 @@ pub struct GradientPrimitiveCpu {
     pub cache_dirty: bool,
     pub gpu_data_address: GpuStoreAddress,
     pub gpu_data_count: i32,
+    pub gpu_blocks: [GpuBlockData; 3],
```

Comment:
I find it a bit inconsistent that some stuff composes `GpuBlockData` on demand while other just keeps `gpu_blocks` locally. If the semantics of `XxxPrimitiveCpu` is supposed to allow easy property manipulation, we should compose `GpuBlockData` on the fly (in `write_gpu_blocks`) for everything. If it's not supposed to be manipulated, then we can have local `[GpuBlockData]` members, and we can replace the `write_gpu_blocks(&self, &mut Vec<>)` into `get_gpu_blocks(&self) -> &[]`.